modules/oe_bootstrap_theme_paragraphs/oe_bootstrap_theme_paragraphs.module

```diff
@@ -329,6 +330,66 @@ function oe_bootstrap_theme_preprocess_paragraph__oe_timeline(array &$variables)
   $variables['hide_from'] = $paragraph->get('field_oe_timeline_expand')->value;
 }
 
+/**
+ * Implements hook_preprocess_paragraph() for paragraph--oe-content-row--variant-inpage-navigation.html.twig.
+ */
+function oe_bootstrap_theme_preprocess_paragraph__oe_content_row__variant_inpage_navigation(array &$variables): void {
+  $paragraph = $variables['paragraph'];
+
+  if (!$paragraph->get('field_oe_title')->isEmpty()) {
+    $variables['title'] = $paragraph->get('field_oe_title')->value;
+  }
+
+  $variables['full_layout'] = TRUE;
+  $variables['scrollspy_id'] = 'bcl-inpage-navigation';
+
+  $links = [];
+  $content = [];
+
+  foreach ($paragraph->get('field_oe_paragraphs')->referencedEntities() as $delta => $sub_paragraph) {
+    /** @var \Drupal\paragraphs\Entity\Paragraph $sub_paragraph */
+    if ($sub_paragraph->hasField('field_oe_title') && !$sub_paragraph->get('field_oe_title')->isEmpty()) {
+      $links_label = $sub_paragraph->get('field_oe_title')->first()->value;
+    }
+    elseif ($sub_paragraph->hasField('field_oe_text') && !$sub_paragraph->get('field_oe_text')->isEmpty()) {
+      $links_label = $sub_paragraph->get('field_oe_text')->first()->value;
+    }
+    elseif ($sub_paragraph->hasField('field_oe_plain_text_long') && !$sub_paragraph->get('field_oe_plain_text_long')->isEmpty()) {
+      $links_label = $sub_paragraph->get('field_oe_plain_text_long')->first()->value;
+    }
+    elseif ($sub_paragraph->bundle() == 'oe_accordion') {
+      $accordion_items = $sub_paragraph->get('field_oe_paragraphs')->referencedEntities();
+      $first_accordion = $accordion_items[0];
+      $links_label = $first_accordion->get('field_oe_text')->first()->value;
+    }
+
+    $unique_id = Html::getUniqueId('bcl-inpage-item-' . $sub_paragraph->id());
+    // Get sub-paragraph translation.
+    $sub_paragraph = \Drupal::service('entity.repository')
+      ->getTranslationFromContext($sub_paragraph, $paragraph->language()->getId());
+    $entity_type = 'paragraph';
+    $view_mode = 'default';
```

Comment:
$view_mode is used only once, no point in holding the memory, u can pass it as a static string